Detemplate 3 Bronze Problems

Shad0wSeven opened the pull request:
Worked on removing templates from the three non Benq templates from [This Thread](https://github.com/cpinitiative/usaco-guide/issues/806#issuecomment-784260077):

- The Cow Signal
- Speeding
- Milk Factory

dongliu0426 reviewed (changes requested):
code should not have loop macros and should not use different macros/typedefs described [here](https://github.com/cpinitiative/usaco-guide/issues/806#issue-811612252)

Shad0wSeven commented:
Will do this rn

maggie-j-liu commented on solutions/USACO dec15-speeding-ticket.mdx:
```diff
@@ -77,26 +30,26 @@ int main(){
+	for(int i = 0; i < N; i++){
 		int length, speed; cin >> length >> speed;
-		For(j, start, start+length){
+		for(int j = start; i < start+length; j++){
```
```suggestion
		for(int j = start; j < start+length; j++){
```

maggie-j-liu commented on solutions/USACO dec15-speeding-ticket.mdx:
```diff
@@ -77,26 +30,26 @@ int main(){
+	for(int i = 0; i < M; i++){
 		int length, speed; cin >> length >> speed;
-		For(j, start, start+length){
+		for(int j = start; i < start+length; j++){
```
```suggestion
		for(int j = start; j < start+length; j++){
```

maggie-j-liu commented on solutions/USACO dec16-cowsignal.mdx:
```diff
@@ -94,11 +28,11 @@ int main(){
-		For(it, 0, K){
-			For(j, 0, N){
-				For(k, 0, K){
+		for(int it = 0; i < K; it++){
```
```suggestion
		for(int it = 0; it < K; it++){
```

thecodingwizard commented on solutions/USACO dec16-cowsignal.mdx:
```diff
@@ -94,11 +28,11 @@ int main(){
-			For(j, 0, N){
-				For(k, 0, K){
+		for(int it = 0; i < K; it++){
+			for(int it = 0; i < N; N++){
```
```suggestion
			for(int j = 0; j < N; j++){
```

maggie-j-liu replied:
```suggestion
			for(int j = 0; j < N; j++){
```

thecodingwizard commented:
Make sure to test your code after te-templatifying it :)

dongliu0426 commented on solutions/USACO dec16-cowsignal.mdx:
```diff
@@ -94,11 +28,11 @@ int main(){
-				For(k, 0, K){
+		for(int it = 0; i < K; it++){
+			for(int it = 0; i < N; N++){
+				for(int k = 0; i < K; k++){
```
```suggestion
				for(int k = 0; k < K; k++){
```

Shad0wSeven commented:
ah crap I was using the same `for(int i = 0; i < n; i++)` and forgot to change alot tysm

jessechoe10 commented:
> Worked on removing templates from the three non Benq templates from [This Thread](https://github.com/cpinitiative/usaco-guide/issues/806#issuecomment-784260077):
> 
> * The Cow Signal
> * Speeding
> * Milk Factory

Is my template really hard to read??? You know they were all mine right 😠 . All kidding aside 😉😆, check to make sure the "detemplifed solution actually compiles" as @thecodingwizard and @dongliu0426 said.

jessechoe10 commented on solutions/USACO dec15-speeding-ticket.mdx:
```diff
@@ -77,26 +30,26 @@ int main(){
 
 	int start = 0;
 	array<int,100> limit;
```
why arr<int, 100> ??? Just make it vector< int >limit(100).

dongliu0426 replied:
i think it's better to use an array than a vector when the object is constant sized.

frodakcin replied:
In general I'd say if it's not bad practice or unclear let them do what they want. Though I should also improve... I tend to make suggestions purely based on preference as well.

jessechoe10 replied:
> In general I'd say if it's not bad practice or unclear let them do what they want. Though I should also improve... I tend to make suggestions purely based on preference as well.

Likewise. I generally prefer vectors LOL.

jessechoe10 replied:
Since it's constant sized can you make it `int limit[100]`?

jessechoe10 commented on solutions/USACO dec15-speeding-ticket.mdx:
```diff
@@ -77,26 +30,26 @@ int main(){
 	}
 
 	start = 0;
 	array<int,100> bessie;
```
vector< int > bessie(100);

Shad0wSeven replied:
see the above comment by frodakcin

jessechoe10 replied:
`int bessie[100];`

jessechoe10 commented:
Can we keep the templified loops??? Why is it that you can remove my template but not Benq's. He uses templified loops as well....

jessechoe10 reviewed (approved):
Everything looks good! I think this PR is ready to be merged.

HarryWangATX commented on solutions/USACO dec15-speeding-ticket.mdx:
```diff
@@ -76,27 +29,27 @@ int main(){
+	for(int i = 0; i < N; i++){
 		int length, speed; cin >> length >> speed;
-		For(j, start, start+length){
+		for(int j = start; j < start+length; j++){
```
```suggestion
		for(int j = start; j < start + length; j++){
```

HarryWangATX commented on solutions/USACO dec15-speeding-ticket.mdx:
```diff
@@ -76,27 +29,27 @@ int main(){
+	for(int i = 0; i < M; i++){
 		int length, speed; cin >> length >> speed;
-		For(j, start, start+length){
+		for(int j = start; j < start+length; j++){
```
```suggestion
		for(int j = start; j < start + length; j++){
```

HarryWangATX commented on solutions/USACO open19-milk-factory.mdx:
```diff
@@ -93,31 +93,22 @@ class Graph{
-#define FOR(i, a, b) for(ll i=a; i<(ll)b; ++i)
-#define ROF(i, a, b) for(ll i=a-1; i>=b; --i)
-#define TRAV(a, x) for(auto& a: x)
+
```
```suggestion
```

maggie-j-liu commented on solutions/USACO dec16-cowsignal.mdx:
```diff
@@ -94,11 +28,11 @@ int main(){
 	int M, N, K; cin >> M >> N >> K;
 
-	For(i, 0, M){
+	for(int i = 0; i < M; i++) {
```
```suggestion
	for(int i = 0; i < M; i++){
```

HarryWangATX commented on solutions/USACO open19-milk-factory.mdx:
```diff
@@ -140,12 +131,12 @@ bool allVisited(){
 	setIO("factory");
 	cin >> n;
-	FOR(i, 0, n-1){
+	for(int i = 0; i < (n-1); i++){
```
```suggestion
	for(int i = 0; i < (n - 1); i++){
```

maggie-j-liu commented on solutions/USACO open19-milk-factory.mdx:
```diff
@@ -93,31 +93,22 @@ class Graph{
-#define sor(x) sort(begin(x), end(x))
 
-void setIO(string name="") { // name is nonempty for USACO file I/O
+void setIO(string name="") { // FastIO See General -> Input/Output
```
```suggestion
void setIO(string name=""){ // FastIO See General -> Input/Output
```

maggie-j-liu commented on solutions/USACO dec15-speeding-ticket.mdx:
```diff
@@ -18,54 +18,7 @@ author: Jesse Choe
-// Input/Output - s is basically the file name without the ".in" and ".out"
-
-void setIO(string s) {
+void setIO(string s) { // FastIO See General -> Input/Output
```
```suggestion
void setIO(string s){ // FastIO See General -> Input/Output
```

maggie-j-liu commented on solutions/USACO dec16-cowsignal.mdx:
```diff
@@ -11,76 +11,10 @@ author: Benjamin Qi, Jesse Choe
-// Input/Output - s is basically the file name without the ".in" and ".out"
-
-void setIO(string name = "") { // name is nonempty for USACO file I/O
+void setIO(string name = "") { // FastIO see General -> Fast Input and Output
```
```suggestion
void setIO(string name = ""){ // FastIO see General -> Fast Input and Output
```

maggie-j-liu reviewed (commented):
make spacing more consistent

darren-yao commented:
Briefly skimmed it, looks detemplated. Reviewers, please check it over again.

dongliu0426 commented on solutions/USACO open19-milk-factory.mdx:
```diff
@@ -93,31 +93,21 @@ class Graph{
-void setIO(string name="") { // name is nonempty for USACO file I/O
+void setIO(string name=""){ // FastIO See General -> Input/Output
     ios_base::sync_with_stdio(0); cin.tie(0); 
     if((ll)name.size() > 0){
```
would prefer if you used `int`s instead

dongliu0426 commented on solutions/USACO open19-milk-factory.mdx:
```diff
@@ -93,31 +93,21 @@ class Graph{
 }
 
-vl adj[101]; vb visited(101); ll n;
+vector<ll> adj[101]; vector<bool> visited(101); ll n;
```
here too

dongliu0426 commented on solutions/USACO open19-milk-factory.mdx:
```diff
@@ -140,12 +130,12 @@ bool allVisited(){
 	cin >> n;
-	FOR(i, 0, n-1){
+	for(int i = 0; i < (n - 1); i++){
 		ll a, b; cin >> a >> b;
```
here too

dongliu0426 commented on solutions/USACO open19-milk-factory.mdx:
```diff
@@ -93,31 +93,21 @@ class Graph{
-vl adj[101]; vb visited(101); ll n;
+vector<ll> adj[101]; vector<bool> visited(101); ll n;
 
 void dfs(ll node){
```
here too (and below)

Shad0wSeven commented:
Removed Long Longs

frodakcin commented:
> Make sure to test your code after te-templatifying it :)

Are these all tested? If so, we can merge, given there's 3 approvals

Shad0wSeven commented:
I haven't tested the latest revision yet will do tmrw after taking USACO :) or if anyone wants to do them but afaik they are untested

frodakcin commented:
> I haven't tested the latest revision yet will do tmrw after taking USACO :) or if anyone wants to do them but afaik they are untested

@Shad0wSeven I'll merge after you test them.

Shad0wSeven commented:
## Milk Factory
![image](https://user-images.githubusercontent.com/19739712/109434577-cd522700-79ca-11eb-878d-23aa415b182c.png)

## Cow Signal
**Note:** Accidentally left in `sz(name)` from the template, removed and replaced with `name.size()` in commit "add size"

![image](https://user-images.githubusercontent.com/19739712/109434683-523d4080-79cb-11eb-9707-4f971af2f51b.png)


## Speeding Ticket
**Note:** Left in `Jesse Choe - Bronze Template` even though there is no template at all but that's not really an issue :)

![image](https://user-images.githubusercontent.com/19739712/109434767-af38f680-79cb-11eb-8ee0-7aa496ff31eb.png)

frodakcin commented:
Aight, you can just verbally confirm you've tested -- don't need images :P

frodakcin commented:
ok "verbally" isnt the right word but you know what i mean
